bin/describe.ml

```diff
@@ -800,10 +860,13 @@ let term : unit Term.t =
       in
       let+ res =
         Build_system.run (fun () ->
-            What.describe what options setup super_context)
+            What.describe what options common setup super_context)
       in
       match res with
-      | Error `Already_reported | Ok None -> ()
+      | Error `Already_reported ->
+        flush_all ();
+        exit 1
```

Comment:
Why do we need this exit and the flush?

Reply:
I can remove them if you want.
I added them to partially fix a behavior I am surprised with: if an error is raised (using `User_error.raise`) then `dune describe` still exits with errno 0. And do not see the standard error in my terminal, although when I redirect it to a file, this file contains the correct error message.
I do not observe this behavior with the released version of `dune`. Has anything changed in between?

Reply:
I can remove this code entirely and use `Build_system.run_exn` instead of `Build_system.run` if you think this is a better solution.

Reply:
Sure. Reported in #6115